crates/walrus-sdk/src/uploader.rs

```diff
@@ -194,91 +210,109 @@ impl DistributedUploader {
 
         let mut requests = WeightedFutures::new(futures);
         let start = Instant::now();
+        let n_shards: usize = self.committees.n_shards().get().into();
+        let cancel_token = cancellation.unwrap_or_default();
 
         let mut blobs_at_quorum = 0;
         let mut results: Vec<NodeResult<R, E>> = Vec::new();
+        let mut cancelled = false;
```

Comment:
Can we avoid this variable by using `.is_cancelled`? https://docs.rs/tokio-util/latest/tokio_util/sync/struct.CancellationToken.html#method.is_cancelled